debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbStreamingChangeEventSource.java

```diff
@@ -248,7 +253,10 @@ else if (rsOffsetContext.lastTimestamp() != null) {
                     }
 
                     try {
-                        pause.pause();
+                        if (noMessageIterations >= THROTTLE_NO_MESSAGE_BEFORE_PAUSE) {
```

Comment:
@jpechane  shouldn't we then emit heart beat event only when this threshold is reached?

Reply:
@jcechace It is IMHO unnecessary optimization.